tools/expotools/src/versioning/android/index.ts

```diff
@@ -0,0 +1,192 @@
+import path from 'path';
+import fs from 'fs-extra';
+import chalk from 'chalk';
+import semver from 'semver';
+import glob from 'glob-promise';
+
+import { getExpoRepositoryRootDir } from '../../Directories';
+
+const EXPO_DIR = getExpoRepositoryRootDir();
+
+function getProjectPaths(abiName: string) {
+  const androidPath = path.join(EXPO_DIR, 'android');
+  const appPath = path.join(androidPath, 'app');
+  const expoviewPath = path.join(androidPath, 'expoview');
+  const versionedAbisPath = path.join(androidPath, 'versioned-abis');
+  const versionedExpoviewAbiPath = path.join(versionedAbisPath, `expoview-${abiName}`);
+  const expoviewBuildGradlePath = path.join(expoviewPath, 'build.gradle');
+  const appManifestPath = path.join(appPath, 'src', 'main', 'AndroidManifest.xml');
+  const templateManifestPath = path.join(EXPO_DIR, 'template-files', 'android', 'AndroidManifest.xml');
+  const settingsGradlePath = path.join(androidPath, 'settings.gradle');
+  const appBuildGradlePath = path.join(appPath, 'build.gradle');
+  const buildGradlePath = path.join(androidPath, 'build.gradle');
+  const sdkVersionsPath = path.join(androidPath, 'sdkVersions.json');
+  const rnActivityPath = path.join(expoviewPath, 'src/main/java/host/exp/exponent/experience/MultipleVersionReactNativeActivity.java');
+  const expoviewConstantsPath = path.join(expoviewPath, 'src/main/java/host/exp/exponent/Constants.java');
+  const testSuiteTestsPath = path.join(appPath, 'src/androidTest/java/host/exp/exponent/TestSuiteTests.java');
```

Comment:
I guess it's a matter of style, but imho either
- making simple funcs like `const testSuiteTestsPath = androidDir => path.join(androidDir, 'src/androidTest/java/host/exp/exponent/TestSuiteTests.java');` or
- moving this whole line to the only dependent function

would make the script cleaner. This way it looks like some god-configuration object.

Reply:
Yeah, initially this function was small but then it grew up... Would it be better if these paths are just a constants defined in the top-level scope? Maybe except one path, that needs to be a function since it needs to know `abiName`.